docs/ga-roadmap.md

```diff
@@ -0,0 +1,173 @@
+# Collector GA Roadmap
+
+This document defines the roadmap followed by the OpenTelemetry Collector,
+along with tentative dates and requirements for GA (stability).
+
+In this document, the term “OpenTelemetry Collector packages" refers to all the golang
+modules and packages that are part of the “OpenTelemetry Collector” ecosystem which
+include the [core](https://github.com/open-telemetry/opentelemetry-collector) and
+[contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib).
+
+In this document, the terms "OpenTelemetry Collector" and "Collector" both specifically
+refer to the entire OpenTelemetry Collector ecosystem’s including core and contrib.
+These terms do not refer to the specification or the Client libraries in this document.
+
+In this document, the term “KR” refers to key results.
+
+## Current Status
+
+The OpenTelemetry Collector ecosystem right now has a lot of packages that are in different
+stages of stability (experimental, alpha, beta, etc.). All these packages have different
+public APIs/Interfaces (e.g. code API, configuration, etc.).
+
+A significant amount of legacy code was inherited from the Collector's ancestor
+[OpenCensus Service](https://github.com/census-instrumentation/opencensus-service), since then
+the Collector changed the internal data model and other significant changes were made.
+
+Trying to mark the entire ecosystem GA, at the same moment, will be a significant effort and
+will take a significant amount of time.
+
+## Proposal
+
+This document proposes a GA Roadmap based on multiple phases, where different parts of the
+collector will be released as stable at different moments of time.
+
+At this moment we are completely defining only the first two phases of the process, and the
+next phases will be defined at a later stage once the Collector maintainers will have
+better understanding of the process and implications.
+
+When other signals data models (proto definition) will be marked as stable, if we follow the
+proposal to just mark things as experimental, the amount of work necessary to stabilize
+them will be minimal: `pdata` is auto-generated so all changes that we do for trace will
+apply to all of them, `consumer` is minimal interface, `component` as well.
+
+Metrics components such as (`receiver/prometheus`, `exporter/prometheusremotewrite`) are
+explicitly left out because metrics data model is not complete, but if that work finishes,
+we can discuss and add them to the Phase 3, or later.
+
+### Phase 1
+
+**Tentative Date:** 2021-03-31
```

Comment:
Is this dependent on the Metrics Spec being first stabilized?

Reply:
It is not -- this is about stable APIs in the Collector, not support for data sources. Phase 2 adds initial data source support and is for tracing only. Phase 3 or later would include metrics.

Reply:
Added one more explicit text in the proposal section that we currently focus only on tracing.